scripts/k8conduit.sh

```diff
@@ -0,0 +1,218 @@
+# - confirm before running if interactive, a flag to run without confirmation?
+#
+
+Help()
+{
+   echo
+   echo "Script to connect to a k8 backing service via an app service"
+   echo
+
+   echo "Syntax:"
+   echo "   k8conduit [-h|i] app-name -- command [args]"
+   echo "      Connect to the default database for app-name"
+   echo
+   echo "or k8conduit [-h|i] -d db-name -k key-vault app-name -- command [args]"
+   echo "      Connect to a specific database from app-name"
+   echo "      Requires a secret containing the DB URL in the specified Azure KV,"
+   echo "      with name {db-name}-database-url"
+   echo
+   echo "options:"
+   echo "   d            database name, required if connecting to a db other than the app default."
+   echo "   i            input file for a restore. Only valid for command psql"
+   echo "   k            key vault that holds the DB URL."
+   echo "   h            print this help."
+   echo
+   echo "parameters:"
+   echo "   app-name     app name to connect to."
+   echo "   command      command to run."
+   echo "                  valid commands are psql, pg_dump, redis-cli"
+   echo "   args         args for the command"
+}
+
+InitSetup()
+{
+   if [ "${RUNCMD}" != "psql" ] && [ "${RUNCMD}" != "pg_dump" ] && [ "${RUNCMD}" != "redis-cli" ]; then
+      echo "Error: invalid command ${RUNCMD}"
+      echo "Only valid options are psql, pg_dump or redis-cli"
+      exit
+   fi
+
+   # If an input file is given, check it exists and is readable
+   if [ "${Inputfile}" != "" ] && [ ! -r "${Inputfile}" ]; then
+      echo "Error: invalid input file"
+      exit
+   fi
+
+   # Set some defailt vars
+   TMOUT=28800 # 8 hour timeout
+   DB_PORT=5432
+   REDIS_PORT=6379
+}
+
+
+CheckInstance()
+{
+   if [ "$INSTANCE" = "" ]; then
+	   echo "Error: Must provide instance name as parameter e.g. apply-qa, apply-review-1234"
+	   exit
+   fi
+   # make sure it's LC
+   INSTANCE=$(echo "${INSTANCE}" | tr '[:upper:]' '[:lower:]')
+   # Lets check the container exists and we can connect to it first
+   if ! kubectl -n development exec -i deployment/"${INSTANCE}" -- echo; then
+      exit
+      fi
+}
+
+SetPorts()
+{
+   # Get a random DEST port for the k8 container
+   # so there is minimal conflict between users
+   DEST_PORT=0
+   until [ $DEST_PORT -gt 1024 ]; do
+	   DEST_PORT=$RANDOM
+   done
+
+   # Get a random LOCAL port
+   # so we can have more than one session if wanted
+   LOCAL_PORT=0
+   until [ $LOCAL_PORT -gt 1024 ]; do
+	   LOCAL_PORT=$RANDOM
+	   nc -z 127.0.0.1 $LOCAL_PORT 2> /dev/null && LOCAL_PORT=0 # try again if it's in use
+   done
+}
+
+SetDBpsql()
+{
+   PORT=${DB_PORT}
+   # Get DB settings
+   # Either from the app DATABASE_URL or the AZURE KV secret
+   if [ "${DBName}" = "" ]; then
+      K8_URL=$(echo 'echo $DATABASE_URL' | kubectl -n development exec -i deployment/"${INSTANCE}" -- sh)
+      DB_URL=$(echo "${K8_URL}" | sed "s/@[^~]*\//@127.0.0.1:${LOCAL_PORT}\//g")
```

Comment:
Maybe give some examples of ${K8_URL} shape to help understand the regex expressions

Reply:
Added example format